pkg/controller/job/indexed_job_utils.go

```diff
@@ -201,6 +201,28 @@ func (oi orderedIntervals) has(ix int) bool {
 	return oi[hi].First <= ix
 }
 
+func (oi orderedIntervals) isProperSubsetOf(superIntervals orderedIntervals) (int, bool) {
+	if len(superIntervals) == 0 && len(oi) != 0 {
+		return oi.total(), false
+	}
+
+	var diff int
+	for _, sub := range oi {
+		if sub.First == sub.Last {
+			if !superIntervals.has(sub.First) {
+				diff++
+			}
+			continue
+		}
+		for element := sub.First; element > sub.Last; element++ {
+			if !superIntervals.has(element) {
```

Comment:
This seems `nlog(n)` if both sets of indexes have n intervals, right? I think we can have a linear algorithm, by walking both lists with two indexes.

Reply:
> This seems nlog(n) if both sets of indexes have n intervals, right?

Yea, I think so, too, since `has` uses the binary search.

>  I think we can have a linear algorithm, by walking both lists with two indexes.

Actually, I tried to implement the way using liner search, but I selected this way to improve readability, and O(nlog (n)) isn't the worst computational complexity.
However, I'm open to replacing this way with the way using the liner search.

Let me reimplement this function.

Reply:
> Actually, I tried to implement the way using liner search, but I selected this way to improve readability, 

I see, makes sense. OTOH this is a helper function which should be easy to do a good coverage by unit tests.

Reply:
> OTOH this is a helper function which should be easy to do a good coverage by unit tests.

That makes sense. I should add some unit tests. Thanks.

Reply:
I re-worked this function using the liner algorithm, and I added unit tests.

Reply:
Done.